### PR TITLE
Ensure spans are cleared out when flushed.

### PR DESCRIPTION
**Pull request by iantalarico**

### CODE REVIEW

**Comment by evildour** [comment]
Thinking about this some more, is this ok? One of the receivers is a timed buffer, right? So it will get a trace proto instance which it holds onto for later, but when it later goes to process it, the spans would have already been cleared out and possibly replaced. If that's the case, it almost seems like we would need to release the reference to the trace proto once it is sent to the consumer and create a new one for subsequent spans.

**Comment by iantalarico** [comment]
Yes you are correct, fixed.  Thanks!

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs [hunk 1]
```diff
@@ -241,9 +241,9 @@ private void CheckStackNotEmpty()
 
         private void Flush()
         {
-            _consumer.Receive(new[] { _trace });
-            // Clear out the flushed spans.
-            _trace.Spans.Clear();
+            var old = _trace;
+            _trace = new TraceProto { TraceId = old.TraceId };
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs]: This looks like it should also copy over a project Id.

**Comment by iantalarico** [comment]
Yea I agree the current API is not ideal.  I think I'll scrap this PR and create a new one with these changes

**Pull request closed without merging**